### PR TITLE
Add wrapper expression type

### PR DESCRIPTION
**Pull request by TheLimeGlass**
### Description
Adds ExpressionType.WRAPPER which should be used for expressions that extend WrapperExpression, aswell as making the fields in ExpressionInfo private in favour of the un-used getter methods of that class. This is mainly for order sorting and in the future if we need to utilize expressions by type. Also made the ExpressionType not null as that makes sense, constructor for nullifying it was never used.

That's all that is the core of this pull request (ExpressionType.class addition and ExpressionInfo.class). The rest is just code standard updates for things touched. Also removed the long deprecated NORMAL type.

Knowledge base:
- A wrapper expression is like an advanced converter. When converting an expression requires multiple inputs to convert. Example being a location using the direction to get another location or a block. Essentially when a DefaultConverter just won't do.
- The order of the enums in the ExpressionType.class is the order in which they're first grabbed.

The order here is to keep in standard with event values. Similarly here; Skript will grab the wrapper expressions after simple, combined and property. In event values the converter is always last in the order. (Of course pattern match everything should be last here though like it is)

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/base/WrapperExpression.java [hunk 1]
```diff
@@ -37,115 +37,112 @@
...
-	
+
 	private Expression<? extends T> expr;
-	
-	@SuppressWarnings("null")
+
 	protected WrapperExpression() {}
-	
+
 	public WrapperExpression(final SimpleExpression<? extends T> expr) {
```
Comment (kiip1) [inline src/main/java/ch/njol/skript/expressions/base/WrapperExpression.java:47]:
```suggestion
	public WrapperExpression(SimpleExpression<? extends T> expr) {
```

**Review by kiip1 — APPROVED** [review]

**Review by Moderocky — APPROVED** [review]

**Review by Moderocky — CHANGES_REQUESTED** [review]
This has conflicts that need fixing

**Review by APickledWalrus — APPROVED** [review]
Looks good. Conflicts need resolved first.

**Comment by sovdeeth** [comment]
Closing due to inactivity

**Pull request closed without merging**